src/us/kbase/workspace/test/database/UtilTest.java

```diff
@@ -0,0 +1,222 @@
+                    "Must provide one and only one of %s name (was: %s) or id (was: %s)",
+                    TYPE_NAME, OK_STRING, OK_LONG)));
+        }
+    }
+
+    @Test
+    public void passIsNullOrEmpty() throws Exception {
+        for (String empty : emptyStrings) {
+            assertThat(INCORRECT_NULL_EMPTY, Util.isNullOrEmpty(empty), is(true));
+        }
+        assertThat(INCORRECT_NULL_EMPTY, Util.isNullOrEmpty(NULL_STRING), is(true));
+    }
+
+    @Test
+    public void failIsNullOrEmpty() throws Exception {
+        for (String nullOrEmpty : nonEmptyStrings) {
+            assertThat(INCORRECT_NULL_EMPTY, Util.isNullOrEmpty(nullOrEmpty), is(false));
+        }
+    }
+
+    @Test
+    public void passNonNull() throws Exception {
+        for (String empty : emptyStrings) {
+            Util.nonNull(empty, OK_STRING);
+        }
+
+        for (String nonEmpty : nonEmptyStrings) {
+            Util.nonNull(nonEmpty, OK_STRING);
+        }
+    }
+
+    @Test
+    public void failNonNull() throws Exception {
+        try {
+            Util.nonNull(null, OK_STRING);
+            fail(EXP_EXC);
+        } catch (Exception got) {
+            TestCommon.assertExceptionCorrect(got, new NullPointerException(OK_STRING));
+        }
+    }
+
+    @Test
+    public void passNoNulls() throws Exception {
+        Util.noNulls(nonEmptyStrings, TYPE_NAME);
+        Util.noNulls(emptyStrings, TYPE_NAME);
+    }
+
+    @Test
+    public void failNoNulls() throws Exception {
+        final List<List<String>> testList = new ArrayList<>();
+        testList.add(emptyStringsWithNull);
+        testList.add(nonEmptyStringsWithNull);
+
+        for (List<String> testListItem : testList) {
+            try {
+                Util.noNulls(testListItem, OK_STRING);
+                fail(EXP_EXC);
+            } catch (Exception got) {
+                TestCommon.assertExceptionCorrect(got, new NullPointerException(OK_STRING));
+            }
+        }
+    }
+
+    @Test
+    public void passCheckNoNullsOrEmpties() throws Exception {
+        Util.checkNoNullsOrEmpties(nonEmptyStrings, TYPE_NAME);
+    }
+
+    @Test
+    public void failCheckNoNullsOrEmpties() throws Exception {
+        final List<List<String>> testList = new ArrayList<>();
+        testList.add(emptyStrings);
+        testList.add(emptyStringsWithNull);
+        testList.add(nonEmptyStringsWithNull);
+
+        for (List<String> testListItem : testList) {
+            try {
+                Util.checkNoNullsOrEmpties(testListItem, TYPE_NAME);
+                fail(EXP_EXC);
+            } catch (Exception got) {
+                TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+                        "Null or whitespace only string in collection " + TYPE_NAME));
+            }
+        }
+    }
+
+    @Test
+    public void passCheckString() throws Exception {
+        final String INCORRECT_CHECKSTRING = "incorrect checkString";
```

Comment:
One thing this test doesn't check is unicode handling - if the codepoint based length check were switched to a `.length()` based check the tests wouldn't detect it, but the behavior would change

Reply:
added some fun unicode characters to the mix

Reply:
:+1: